### PR TITLE
[WIP] Void Support

### LINKED ISSUE #81: Void headset support
I'm putting this in a separate issue so that we don't pollute #52.

From @aldistuck:
[CorsairUSBVoid.pcapng.gz](https://github.com/ckb-next/ckb-next/files/1709461/CorsairUSBVoid.pcapng.gz)
I attempted to use Wireshark to capture usb communication from my Corsair usb Void headset (corded). From the start the UI could see usb connections. One of them contained the headset so I just started a capture on that so I'm assuming everything in this is only between the headset and what the capture was calling the host. Nothing began happening until I opened the CUE 2 software (captured this on Windows 10). Once CUE 2 was open I started seeing data in Wireshark. I went to the settings for the headset and cycled through all the settings for lighting effects, EQ, and infomic. I hope this is helpful information.

[CorsairUSBVoid-0.pcapng.gz](https://github.com/ckb-next/ckb-next/files/1709571/CorsairUSBVoid-0.pcapng.gz)
[CorsairUSBVoid-1.pcapng.gz](https://github.com/ckb-next/ckb-next/files/1709572/CorsairUSBVoid-1.pcapng.gz)
Here are a couple more traces I recorded. I opened up a youtube video to get audio going through the headset. The -1 capture seems to have an additional source I did not have prior.

### PR DESCRIPTION
**Pull request by SethBarberee**
Supersedes #528 

- [X] Headset recognized by daemon
- [X] Headset recognized by GUI
- [ ] RGB Control <- Getting errors when sending the message so working on that (uncomment the usbsend and it'll stall)

Fixes #81 

If you wanna test if the daemon picks up your headset, please do and let me know if I need to add more usbid's for respective headset

### CODE REVIEW

File: src/daemon/device_void.c [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        { 0xcb, 0x03, 0x1e, 0x00, 0x1b, 0x00, 0x1a, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 }
+    };
+
+    uchar init17[2] =
+        { 0xc9, 0x66 };
+
+    if(!usbsend(kb, init1, 1))
+        return -1;
+
+    if(!usbsend(kb, init2[0], 2))
```
Comment (SethBarberee) [inline src/daemon/device_void.c:92]: Currently, void breaks here with broken pipe.. Wanna do a check on commands later.

**Comment by SethBarberee** [comment]
Current status: 
```
* Send active command on freshly plugged void
* Void turns rgb off
* Send rgb command and displays color
* Send idle command to return to default rainbow
* Send active command and void does blue blink animation
* RGB commands won't work anymore but active/idle work
```

**Comment by SethBarberee** [comment]
Hardcoded the endpoint path. This fixed rgb not working after an active/idle cycle. Mode is still blue blinking after a cycle but an rgb command will work now.

Mute indicator only works in hardware mode, not software mode.

Mute button press
Messages don't change if void is muted or not or hardware/software mode
```
[I] ckb1 Input Recv: 64 02 00 10 00 
[E] process_input_urb (keymap.c:374): Unknown data received in input thread 64 from endpoint 84

```

Mute button release
```
[I] ckb1 Input Recv: 64 00 00 10 00 
[E] process_input_urb (keymap.c:374): Unknown data received in input thread 64 from endpoint 84
```

**Comment by thespyboss** [comment]
Is there anything I can do to help? I have a Wireless Void Headset that i can use to help with development. I know before this whole thing got cleared out, you needed packets from Wireshark monitoring the headset and ICUE. Is that still needed?

**Comment by SethBarberee** [comment]
That would be appreciated just to see how different they are from the wired version.

**Comment by thejacer87** [comment]
just got this headset. how can i help? do i need to pull this pr and rebuild the software?

**Comment by nomis6432** [comment]
I have a corsaire virtuose SE. Would you be interested in the packets from wireshark for this headset?

**Comment by SethBarberee** [comment]
This is mainly for Void however it would be interesting to see the packets for differences. 

Working on possibly buying a void wireless for christmas

**Comment by LordMZTE** [comment]
Is support for the Corsair Virtuoso headset planned too?

**Comment by zearthur99** [comment]
I don't have any C experience but I would love to help. I have a Void Elite USB.

**Comment by rlscipio** [comment]
Can we have an Update? I'm happy to contribute to this project by testing the programm and providing logs btw just hmu

**Comment by tatokis** [comment]
Still waiting on @SethBarberee.

**Comment by vrooze** [comment]
Not trying to necro an old one; however...

I'm also waiting on having the Void supported in ckb :)

Now I am a developer/techy (not with C though :cry: ) who dual boots Windows and Linux daily so if you need a wireshark log I can provide from either OS.

Cheers!

**Comment by thejacer87** [comment]
Same boat as @vrooze... Can dual boot if necessary to help out. Would just need instructions on how to do it to help out.

Unlike vrooze, I DO want to bring this back from the dead lol

**Comment by SethBarberee** [comment]
> Still waiting on @SethBarberee.

Sorry but I don't have the time to continue this. I've been awfully busy at work. I hope this gives enough of a platform for the future.

**Pull request closed without merging**